aten/src/ATen/native/native_functions.yaml

```diff
@@ -5142,34 +5142,40 @@
+- func: conv_transpose2d(Tensor self, Tensor weight, int[2] kernel_size, Tensor? bias=None, int[2] stride=1, int[2] padding=0, int[2] output_padding=0, int[2] dilation=1, *, Tensor(a!) out) -> Tensor(a!)
   python_module: nn
+  dispatch:
+    CPU: conv_transpose2d_out_cpu
+    CUDA: legacy::cuda::_thnn_conv_transpose2d_out
 
-- func: thnn_conv_transpose2d(Tensor self, Tensor weight, int[2] kernel_size, Tensor? bias=None, int[2] stride=1, int[2] padding=0, int[2] output_padding=0, int[2] dilation=1) -> Tensor
+- func: conv_transpose2d(Tensor self, Tensor weight, int[2] kernel_size, Tensor? bias=None, int[2] stride=1, int[2] padding=0, int[2] output_padding=0, int[2] dilation=1) -> Tensor
   python_module: nn
+  dispatch:
+    CPU: conv_transpose2d_cpu
+    CUDA: legacy::cuda::_thnn_conv_transpose2d
 
-- func: thnn_conv_transpose2d_forward(Tensor self, Tensor weight, int[2] kernel_size, Tensor? bias, int[2] stride, int[2] padding, int[2] output_padding, int[2] dilation, *, Tensor(a!) output, Tensor(b!) columns, Tensor(c!) ones) -> (Tensor(a!), Tensor(b!), Tensor(c!))
+- func: conv_transpose2d_forward(Tensor self, Tensor weight, int[2] kernel_size, Tensor? bias, int[2] stride, int[2] padding, int[2] output_padding, int[2] dilation, *, Tensor(a!) output, Tensor(b!) columns, Tensor(c!) ones) -> (Tensor(a!), Tensor(b!), Tensor(c!))
   python_module: nn
   dispatch:
-    CPU: legacy::cpu::_thnn_conv_transpose2d_forward_out
+    CPU: conv_transpose2d_forward_out_cpu
     CUDA: legacy::cuda::_thnn_conv_transpose2d_forward_out
 
-- func: thnn_conv_transpose2d_forward(Tensor self, Tensor weight, int[2] kernel_size, Tensor? bias, int[2] stride, int[2] padding, int[2] output_padding, int[2] dilation) -> (Tensor output, Tensor columns, Tensor ones)
+- func: conv_transpose2d_forward(Tensor self, Tensor weight, int[2] kernel_size, Tensor? bias, int[2] stride, int[2] padding, int[2] output_padding, int[2] dilation) -> (Tensor output, Tensor columns, Tensor ones)
   python_module: nn
   dispatch:
-    CPU: legacy::cpu::_thnn_conv_transpose2d_forward
+    CPU: conv_transpose2d_forward_cpu
     CUDA: legacy::cuda::_thnn_conv_transpose2d_forward
 
-- func: thnn_conv_transpose2d_backward(Tensor grad_output, Tensor self, Tensor weight, int[2] kernel_size, int[2] stride, int[2] padding, int[2] output_padding, int[2] dilation, Tensor columns, Tensor ones, *, Tensor?(a!) grad_input, Tensor?(b!) grad_weight, Tensor?(c!) grad_bias) -> (Tensor(a!), Tensor(b!), Tensor(c!))
+- func: conv_transpose2d_backward(Tensor grad_output, Tensor self, Tensor weight, int[2] kernel_size, int[2] stride, int[2] padding, int[2] output_padding, int[2] dilation, Tensor columns, Tensor ones, *, Tensor?(a!) grad_input, Tensor?(b!) grad_weight, Tensor?(c!) grad_bias) -> (Tensor(a!), Tensor(b!), Tensor(c!))
   python_module: nn
   dispatch:
-    CPU: legacy::cpu::_thnn_conv_transpose2d_backward_out
+    CPU: conv_transpose2d_backward_out_cpu
```

Comment:
One of the `CI` errors is `'_thnn_conv_transpose2d_backward' is not a member of 'at::native::legacy::cuda'`.

I'm pretty sure that if you do CPU/CUDA separately, you cannot remove `thnn_conv_transpose2d_backward` from `native_functions.yaml` until CUDA is ported as well.

This is why I recommend doing CPU/CUDA together.

Reply:
I agree with @skrah . 
However, to just test on CPU, use `USE_CUDA=0` env variable when building. This allows finishing CPU porting before starting the CUDA porting.

Reply:
@skrah @pearu thanks so much for the feedback! I will try to find a workaround for that ... not sure if it would be a good approach to port cpu and gpu in the same PR. I needed to move some steps back and rewrote again my code ... locally, the problematic tests passed. I will try to fix this GPU problem to move forward this PR.

Reply:
I see .. thanks for the explanation @skrah ! I move steps back and worked on my first code ... and just realized that I already started the porting for cuda ... so I will probably port for gpu for 2d also im2col and col2im .. but I think the comparison between files will be easy ... I am not changing the structure here.